sdk-tests/src/test/java/io/dapr/it/actors/ActorSdkResiliencytIT.java

```diff
@@ -0,0 +1,141 @@
+import io.dapr.actors.client.ActorProxyBuilder;
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.resiliency.ResiliencyOptions;
+import io.dapr.it.BaseIT;
+import io.dapr.it.DaprRun;
+import io.dapr.it.ToxiProxyRun;
+import io.dapr.it.actors.services.springboot.DemoActor;
+import io.dapr.it.actors.services.springboot.DemoActorService;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test SDK resiliency.
+ */
+public class ActorSdkResiliencytIT extends BaseIT {
+
+  private static final ActorId ACTOR_ID = new ActorId(UUID.randomUUID().toString());
+
+  private static final int NUM_ITERATIONS = 20;
+
+  private static final Duration TIMEOUT = Duration.ofMillis(1000);
+
+  private static final Duration LATENCY = TIMEOUT.dividedBy(2);
+
+  private static final Duration JITTER = TIMEOUT.multipliedBy(2);
+
+  private static final int MAX_RETRIES = -1;  // Infinity
+
+  private static DaprRun daprRun;
+
+  private static DaprClient daprClient;
+
+  private static DemoActor demoActor;
+
+  private static ToxiProxyRun toxiProxyRun;
+
+  private static DemoActor toxiDemoActor;
+
+  private static DemoActor resilientDemoActor;
+
+  private static DemoActor oneRetryDemoActor;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    daprRun = startDaprApp(
+            ActorSdkResiliencytIT.class.getSimpleName(),
+            DemoActorService.SUCCESS_MESSAGE,
+            DemoActorService.class,
+            true,
+            60000);
+
+    ActorId actorId = new ActorId(UUID.randomUUID().toString());
+
+    // HTTP client is deprecated, so SDK resiliency is for gRPC client only.
+    daprRun.switchToGRPC();
+    demoActor = buildDemoActorProxy(null);
+    daprClient = new DaprClientBuilder().build();
+
+    toxiProxyRun = new ToxiProxyRun(daprRun, LATENCY, JITTER);
+    toxiProxyRun.start();
+    toxiProxyRun.use();
+    toxiDemoActor = buildDemoActorProxy(new ResiliencyOptions().setTimeout(TIMEOUT));
+    resilientDemoActor = buildDemoActorProxy(
+            new ResiliencyOptions().setTimeout(TIMEOUT).setMaxRetries(MAX_RETRIES));
+    oneRetryDemoActor = buildDemoActorProxy(
+            new ResiliencyOptions().setTimeout(TIMEOUT).setMaxRetries(1));
+  }
+
+  private static DemoActor buildDemoActorProxy(ResiliencyOptions resiliencyOptions) {
+    ActorProxyBuilder<DemoActor> builder =
+            new ActorProxyBuilder(DemoActor.class, newActorClient(resiliencyOptions));
+    return builder.build(ACTOR_ID);
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    if (toxiProxyRun != null) {
+      toxiProxyRun.stop();
+    }
+  }
+
+  @Test
+  @Ignore("Flaky when running on GitHub actions")
```

Comment:
I spent significant amount of time to remove flakiness. I did fix an issue in AbstractActor because of this test and removed flakiness running locally - not sure why it still flaky in GH Actions.